consumer.js

```diff
@@ -0,0 +1,14 @@
+"use strict"
+
+module.exports = function consumer( options ){
+  var seneca = this
+
+  seneca.add('role:npm,info:change',  onNpmChange)
+
+  function onNpmChange (pkg, respond) {
+    var seneca = this
+    seneca.act({role:'info', res:'part', name: pkg.name}, (err) => {
```

Comment:
The pattern here should be `role:info,req:part`. This is where your error may be originating from. The idea is to mimic a call from info (thereby keeping the load off it). Info handles `role:info,res:part` so it is tricked into updating the module cache it has. This should fix your search only updating issue too.